capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -236,4 +244,76 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
+      id = jsonObject.get("id").getAsString();
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
+      System.err.println("This error was likely caused by a lack of an \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    if (id.length() == 0) {
+      System.err.println("Error caused by either an empty or non-existent \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    Map<String, Object> update = new HashMap<>();
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    boolean containsParameter = false;
+    for (String key : jsonObject.keySet()) {
+      for (Field field: fields) {
+        String type = field.getType().getName();
+        if (field.getName().equals(key)) {
+          containsParameter = true;
+          if (type.equals("int") || type.equals("java.lang.Integer")) {
+            update.put(key, jsonObject.get(key).getAsInt());
+          }
+          if (type.equals("java.lang.String") && !field.getName().equals("id")) {
+            update.put(key, jsonObject.get(key).getAsString());
+          }
+          if (type.equals("boolean") || type.equals("java.lang.Boolean")) {
+            update.put(key, jsonObject.get(key).getAsBoolean());
+          }
+          else {
+            System.err.println("Error: Bad type in request: can't be cast to Integer, String or Boolean.");
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return null;
+          }
+        }
+        else if (("add_"+field.getName()).equals(key)) {
+          if (type.equals("java.util.List") || type.equals("java.util.ArrayList")) {
+            containsParameter = true;
+            update.put(field.getName(), FieldValue.arrayUnion(jsonObject.get(key).getAsString()));
+          }
+        }
+        else if (("remove_"+field.getName()).equals(key)) {
+          if (type.equals("java.util.List") || type.equals("java.util.ArrayList")) {
+            containsParameter = true;
+            update.put(field.getName(), FieldValue.arrayRemove(jsonObject.get(key).getAsString()));
+          }
+        }
+      }
+      if (!containsParameter) {
+        System.err.println("Error: The object does not have the field specified in the request parameter.");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return null;
+      }
+      containsParameter = false;
+    }
+
+    ApiFuture<WriteResult> writeResult = collectionReference.document(id).set(update, SetOptions.merge());
+    try {
+      System.out.println("Update time : " + writeResult.get().getUpdateTime());
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return null;
+    }
+    return getById(id, collectionReference, request, response, genericClass).get(0);
```

Comment:
Cool! I like how you're updating the data, then getting the updated object from the database again to return to the user

Reply:
Thanks!